mypy/semanal.py

```diff
@@ -2355,6 +2356,41 @@ def is_classvar(self, typ: Type) -> bool:
     def fail_invalid_classvar(self, context: Context) -> None:
         self.fail('ClassVar can only be used for assignments in class body', context)
 
+    def process_module_assignment(self, s: AssignmentStmt) -> None:
+        """Check if s assigns a module an alias name; if so, update symbol table."""
+        # TODO support module alias assignment not in global scope
+        if not self.is_module_scope():
+            return
+        self._process_module_assignment(s.lvalues, s.rvalue, s)
+
+    def _process_module_assignment(
+            self,
+            lvals: List[Expression],
+            rval: Expression,
+            ctx: AssignmentStmt,
+    ) -> None:
+        """Propagate module references across assignments.
+
+        Recursively handles the simple form of iterable unpacking; doesn't
+        handle advanced unpacking with *rest, dictionary unpacking, etc.
+
+        """
+        if all(isinstance(v, (TupleExpr, ListExpr)) for v in lvals + [rval]):
+            litemlists = [v.items for v in cast(List[Union[TupleExpr, ListExpr]], lvals)]
+            ritems = cast(Union[TupleExpr, ListExpr], rval).items
+            for rv, *lvs in zip(ritems, *litemlists):
+                self._process_module_assignment(lvs, rv, ctx)
```

Comment:
Can you be more specific what you find unclear about this code so I can improve the clarity? When I rewrite this code to not support nesting, it doesn't make it any simpler or clearer. The recursive approach is still the simplest even for a single layer of iterable (otherwise you have to do something else weird to support both iterable and single rval), and once you have recursion, not supporting nesting is more complex than supporting it.

I agree that nested unpacked module assignment is likely very rare, but it seems strange to me to reduce compatibility with true Python semantics in order to achieve (AFAICS) a negligible or nonexistent gain in clarity/simplicity.